src/Raven.Server/Documents/AttachmentsStorage.cs

```diff
@@ -789,6 +797,32 @@ public ByteStringContext.InternalScope GetAttachmentPrefix(DocumentsOperationCon
             return scope;
         }
 
+        public static AttachmentType GetAttachmentTypeByKey(Slice keySlice)
+        {
+            var index = 0;
+            var found = false;
+            for (int i = 0; i < keySlice.Size; i++)
+            {
+                if (Convert.ToChar(keySlice[i]) == SpecialChars.RecordSeparator)
```

Comment:
Hmm.. we have places that we do
 `var idEnd = key.Content.IndexOf(SpecialChars.RecordSeparator);` 

or 
```
var val = bytes[i];
if (val != SpecialChars.RecordSeparator)
```
and I wonder if it is correct.. in case we have some unicode char might we identify it wrong?